util/reggen/ip_block.py

```diff
@@ -348,3 +348,18 @@ def get_signal_by_name_as_dict(self, name: str) -> Dict[str, object]:
         else:
             raise ValueError("Signal {} does not exist in IP block {}"
                              .format(name, self.name))
+
+    def has_shadow_reg(self) -> bool:
```

Comment:
Nit: It's not a big deal, but maybe this should be the same name as the equivalent method on `RegBlock`? (shadow vs. shadow**ed**). Either name looks good to me, but maybe we should be consistent.

Reply:
yep, it should be the same. I think i keep on forgetting which one i used.